spec/unit/harvestdor-indexer_spec.rb

```diff
@@ -38,6 +38,20 @@
     expect(@hdor_client.config.default_set).to eq(@config['harvestdor']['default_set'])
   end
 
+  describe '#resources' do
+    it 'does not persist resources across calls' do
+      VCR.use_cassette('single_rsolr_connection_call') do
+        hdor_client = @indexer.send(:harvestdor_client)
+        allow(@indexer.dor_fetcher_client).to receive(:druid_array).and_return(['druid:yg867hg1375', 'druid:jf275fd6276', 'druid:nz353cp1092', 'druid:tc552kq0798', 'druid:th998nk0722', 'druid:ww689vs6534'])
+
+        a = @indexer.resources.first
+        b = @indexer.resources.first
+
+        expect(a).to_not eq b
+      end
+    end
```

Comment:
should there also be tests in here to show that does something diff for items than for a collection?